### DISCUSSION TITLE
Does rainbowgum support "structured logging"?

### DISCUSSION THREAD

**Discussion by hrstoyanov** [jstachio/rainbowgum#202] [General]
https://itnext.io/deep-dive-into-the-structured-logging-in-spring-boot-3-4-89b2a5b1b75f

**Comment by agentgt** [comment]
I saw this before Spring Boot released and the API is kind of weird but yeah the boot module will hopefully support it.

Rainbow Gum on its own does have a JSON module that helps you make your own JSON programmatically but it is fairly flat because the MDC and key value pairs (SLF4J fluent) are String key and String value.

So it is structured in the sense that it can do `Map<String,String>` (ignoring the normal fields like time and stuff) but not `Map<String,Object>`. 

This is largely because before SLF4J fluent the only way to add structure was through the MDC but the MDC only supports String values.

Anyway I do plan on looking into improving this.

**Reply by hrstoyanov** [reply]
Ok, I see. Note that the Spring strutured logging adheres to 3 different "standard formats" .

Side note: I ma trying to avoid MDC and other local thread solutions, now that we have ScopedValue in Java
